gpytorch/means/multitask_mean.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from gpytorch.means import Mean
+
+
+class MultitaskMean(Mean):
+    def __init__(self, base_means, n_tasks):
+        super(MultitaskMean, self).__init__()
+
+        if isinstance(base_means, Mean):
+            base_means = [base_means]
+
+        if not isinstance(base_means, list) or (len(base_means) != 1 and len(base_means) != n_tasks):
+            raise RuntimeError("base_means should be a list of means of length either 1 or n_tasks")
+
+        if len(base_means) == 1:
+            base_means = base_means * n_tasks
```

Comment:
It's a trap! This will not actually instantiate new mean objects, but put references to the same object into a list, so the different tasks would mechanically share the same parameters. Instead we;d have do something like this:

```
base_means = base_means + [deepcopy(base_means[0] for i in range(n_tasks - 1)]
```
I guess we'd have to put the means into a ModuleList then.

Reply:
I was thinking the default behavior was to have a constant mean shared across tasks if the user only specifies one mean, otherwise they pass in a list of different ones.

Should the default behavior instead be to have one mean per task? I guess that makes sense. We'll also need to use a ModuleList either way, though.

Reply:
Yeah I think there should be one mean per task, in the basic case of a constant mean it seems that this would effectively amount to a bias kernel. With fancier mean functions I guess it's up to the user to do something reasonable.